src/wasm_emulation/contract.rs

```diff
@@ -58,27 +61,22 @@ fn apply_storage_changes<ExecC>(storage: &mut dyn Storage, output: &WasmRunnerOu
 /// number of contract executions and queries on one instance. For this reason it is significatly
 /// higher than the limit for a single execution that we have in the production setup.
 const DEFAULT_GAS_LIMIT: u64 = 500_000_000_000_000; // ~0.5s
-const DEFAULT_MEMORY_LIMIT: Option<Size> = Some(Size::mebi(16));
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct DistantContract {
-    pub contract_addr: Addr,
-}
-
-#[derive(Serialize, Deserialize, Debug, Clone)]
+#[derive(Debug, Clone)]
 pub struct DistantCodeId {
     pub code_id: u64,
+    pub module: (Engine, Module),
```

Comment:
Engine and Modules go together and MUST be used together, otherwise segmentation faults can happen

Reply:
Wow, that's wild Engine that can't be reused with different module